src/Microsoft.AspNet.OData.Shared/Formatter/Serialization/ODataResourceSerializer.cs

```diff
@@ -634,6 +638,105 @@ public virtual ODataResource CreateResource(SelectExpandNode selectExpandNode, R
             }
         }
 
+        /// <summary>
+        /// AppendInstanceAnnotations
+        /// </summary>
+        /// <param name="resource"></param>
+        /// <param name="resourceContext"></param>
+        [SuppressMessage("Microsoft.Performance", "CA1804:RemoveUnusedLocals", MessageId = "edmTypeReference")]
+        public virtual void AppendInstanceAnnotations(ODataResource resource, ResourceContext resourceContext)
+        {
+            IEdmStructuredType structuredType = resourceContext.StructuredType;
+            IEdmStructuredObject structuredObject = resourceContext.EdmObject;
+            object value;
+
+            PropertyInfo instanceAnnotationInfo = EdmLibHelpers.GetInstanceAnnotationsDictionary(structuredType,
+                resourceContext.EdmModel);
```

Comment:
move this before line 651; do your setup first, then declare your output variable immediately before using it. Makes the code easier to follow.

Reply:
#Resolved